app/controllers/users/sessions_controller.rb

```diff
@@ -130,8 +130,10 @@ def cache_active_profile
       begin
         cacher.save(auth_params[:password], profile)
       rescue Encryption::EncryptionError => err
-        profile.deactivate(:encryption_error)
-        analytics.track_event(Analytics::PROFILE_ENCRYPTION_INVALID, error: err.message)
+        if profile
+          profile.deactivate(:encryption_error)
```

Comment:
Interesting that with this change, but no change to the specs, everything passes. I thought about going down this route originally, but the test was looking for a particular decryption error to get thrown, which couldn't come from this code since it gets rescued.

Reply:
The error comes later. See my commit message.